### ISSUE TITLE
webgpu: Add new op to support some classfication models and super-resolution models

### ISSUE THREAD

**Issue by NALLEIN** [tensorflow/tfjs#3405]
Add some operations that are frequently used in classification models and super-resolution models

- batchToSpaceND
- softmax
- avgPool

To get help from the community, we encourage using Stack Overflow and the [`tensorflow.js`](https://stackoverflow.com/questions/tagged/tensorflow.js) tag.

#### Code to reproduce the bug / link to feature request
If you would like to get help from the community, we encourage using Stack Overflow and the [`tensorflow.js`](https://stackoverflow.com/questions/tagged/tensorflow.js) tag.

GitHub issues for this repository are tracked in the [tfjs union repository](https://github.com/tensorflow/tfjs/issues).

Please file your issue there, following the guidance in [that issue template](https://github.com/tensorflow/tfjs/blob/master/ISSUE_TEMPLATE.md).
